app/tests/unit/components/ProgressStepIndicator.test.tsx

```diff
@@ -0,0 +1,53 @@
+import StatusBadgeColor from 'components/helpers/StatusBadgeColor';
+import ProgressStepIndicator from 'components/ProgressStepIndicator';
+import {mount} from 'enzyme';
+import React from 'react';
+
+describe('The Progress Step Indicator', () => {
+  it('Renders with no steps', () => {
+    const componentUnderTest = mount(<ProgressStepIndicator steps={[]} />);
+    expect(componentUnderTest).toMatchSnapshot();
+  });
+
+  it('Matches snapshot with 3 tests', () => {
+    const componentUnderTest = mount(
+      <ProgressStepIndicator
+        steps={[
+          {
+            badgeStyle: StatusBadgeColor.APPROVED,
+            description: 'test step 1',
+            number: 99
+          },
+          {
+            badgeStyle: StatusBadgeColor.APPROVED,
+            description: 'test step 2',
+            number: 100
+          },
+          {
+            badgeStyle: StatusBadgeColor.REJECTED,
+            description: 'test step 3',
+            number: 110
+          }
+        ]}
+      />
+    );
+    expect(componentUnderTest).toMatchSnapshot();
+    expect(componentUnderTest.find('.numberedCircle')).toHaveLength(3);
+    expect(componentUnderTest.find('div.stepDescription')).toHaveLength(3);
+  });
+
+  const stepCountTestCases = [1, 2, 3, 4, 5, 6];
+  test.each(stepCountTestCases)('Generates %p progress items', (stepCount) => {
```

Comment:
Just a thought - but instead of 6 nearly identical test cases, might it be more valuable to have some assertions that the `wrapper.find('.numberedCircle').text()` equals the number (likewise with the descriptions) and that the `className` contains the status color?

Reply:
(I didn't mean the new assertions also need to be repeated 6 times, but good enough for go time! 😛)

Reply:
Haha, this way it cycles through all the input possibilities, we've got full test coverage!